### ISSUE TITLE
AWS Instance - "unknown Torch class <nn.SpatialReflectionPadding>"

### ISSUE THREAD

**Issue by Sugarbank** [junyanz/CycleGAN#17]
I've made an AWS instance for people to test this.
All setup with Torch and dependencies.

However getting this error when running...

ubuntu@ip-172-31-3-22:~/CycleGAN$ DATA_ROOT=./datasets/ae_photos name=style_cezanne_pretrained model=one_direction_test phase=test loadSize=256 fineSize=256 resize_or_crop="scale_width"  th test.lua
------------------- Options -------------------	
                DATA_ROOT: ./datasets/ae_photos	
               align_data: 0	
             aspect_ratio: 1	
                batchSize: 1	
          checkpoints_dir: ./checkpoints	
           continue_train: 1	
                    cudnn: 1	
                  display: 1	
               display_id: 200	
                 fineSize: 256	
                     flip: 0	
                      gpu: 1	
                 how_many: all	
                 input_nc: 3	
                 loadSize: 256	
                    model: one_direction_test	
                 nThreads: 1	
                     name: style_cezanne_pretrained	
                     norm: instance	
                output_nc: 3	
                    phase: test	
               preprocess: regular	
           resize_or_crop: scale_width	
              results_dir: ./results/	
           serial_batches: 1	
                     test: 1	
          which_direction: AtoB	
              which_epoch: latest	
-----------------------------------------------	
GPU Mode	
function: 0x412208d8	
DataLoader UnalignedDataLoader was created.	
Starting donkey with id: 1 seed: 5476
table: 0x41ea8e98
table: 0x41ec0fb8
running "find" on each class directory, and concatenate all those filenames into a single file containing all image paths for a given class
now combine all the files to a single large file
load the large concatenated list of sample paths to self.imagePath
cmd..wc -L '/tmp/lua_kY7Jdn' |cut -f1 -d' '
205 samples found................................... 0/205 ...................................................] ETA: 0ms | Step: 0ms          
Updating classList and imageClass appropriately
 [================================================== 1/1 ====================================================>] ETA: 0ms | Step: 0ms          
Cleaning up temporary files
Dataset Size A: 	205	
Starting donkey with id: 1 seed: 514
table: 0x40c39b98
table: 0x41888090
running "find" on each class directory, and concatenate all those filenames into a single file containing all image paths for a given class
now combine all the files to a single large file
load the large concatenated list of sample paths to self.imagePath
cmd..wc -L '/tmp/lua_eMI6Qv' |cut -f1 -d' '
205 samples found................................... 0/205 ...................................................] ETA: 0ms | Step: 0ms          
Updating classList and imageClass appropriately
 [================================================== 1/1 ====================================================>] ETA: 0ms | Step: 0ms          
Cleaning up temporary files
Dataset Size B: 	205	
use InstanceNormalization	
loading previously trained model (/home/ubuntu/CycleGAN/checkpoints/style_cezanne_pretrained/latest_net_G.t7)	
use InstanceNormalization	
/home/ubuntu/torch/install/bin/luajit: /home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:249: unknown Torch class <nn.SpatialReflectionPadding>
stack traceback:
	[C]: in function 'error'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:249: in function 'readObject'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:271: in function 'readObject'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:271: in function 'readObject'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:257: in function 'readObject'
	/home/ubuntu/torch/install/share/lua/5.1/torch/File.lua:294: in function 'load'
	/home/ubuntu/CycleGAN/util/util.lua:192: in function 'load_test_model'
	./models/one_direction_test_model.lua:23: in function 'Initialize'
	test.lua:80: in main chunk
	[C]: in function 'dofile'
	...untu/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:131: in main chunk
	[C]: at 0x00406670

**Comment by k3nnywilliam** [comment]
Me too, same error except it's on my machine instead of aws.

**Comment by whatru** [comment]
Mine is 
File.lua:343: unknown Torch class <torch.CudaTensor>

How to fix it?

**Comment by k3nnywilliam** [comment]
Update: I managed to get CycleGAN running on Google Cloud with Ubuntu 16.04LTS, CUDA 8.0 and CUDNN 5.1. I'm using free trial at the moment, which is awesome because Google provides free usage of its cloud services for 12 months and also $300 worth of credit. So use wisely.

These steps might help:
-First, get the latest torch repo and reinstall (if you have installed it previously).
-Then do ./update.sh
-After that, do 'luarocks list' to check whether packages such as nngraph, class and display have been installed (if I could remember correctly, only nngraph has been installed by default).

Hope it works for you!

**Comment by junyanz** [comment]
Thanks @k3nnywilliam  for the update!
